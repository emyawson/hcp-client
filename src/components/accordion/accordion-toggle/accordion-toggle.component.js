import * as React from 'react';

import { handleKeyPressClick } from 'src/utils';
import { ArrowButton } from 'src/components';

import {
  AccordionToggleDiv,
  AccordionToggleContentDiv,
  AccordionToggleButtonContainerDiv,
} from '../accordion.style';

export const AccordionToggle = ({
  active = false,
  children,
  disabled = false,
  id,
  isNested = false,
  toggleHandler = () => undefined,
}) => {
  const dispatchEventsIfEnabled = (e: Event) => {
    if (!disabled) {
      toggleHandler(e);
    }
  };
  return (
    <AccordionToggleDiv
      active={active}
      disabled={disabled}
      isNested={isNested}
      onClick={dispatchEventsIfEnabled}
      onKeyPress={handleKeyPressClick(dispatchEventsIfEnabled)}
      tabIndex={0}
      role="heading"
    >
      <AccordionToggleContentDiv id={`AccordionTitle--${id}`}>
        {children}
      </AccordionToggleContentDiv>
      <AccordionToggleButtonContainerDiv active={active} disabled={disabled}>
        <ArrowButton disabled={disabled} direction={'down'} />
      </AccordionToggleButtonContainerDiv>
    </AccordionToggleDiv>
  );
};
