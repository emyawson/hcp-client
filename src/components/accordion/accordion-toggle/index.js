export * from './accordion-toggle.component';
