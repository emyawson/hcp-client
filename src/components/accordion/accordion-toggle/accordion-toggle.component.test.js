import React from 'react';
import { shallow } from 'enzyme';

import { AccordionToggle } from './accordion-toggle.component';

const mockToggleHandler = jest.fn();

describe('Accordion Toggle', () => {
  const mockProps = {
    active: true,
    disabled: false,
    id: 'abc-123',
    toggleHandler: mockToggleHandler,
  };
  it('renders correctly', () => {
    const wrapper = shallow(<AccordionToggle {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('dispatches a toggle event only when enabled', () => {
    const disabledProps = {
      ...mockProps,
      disabled: true,
    };
    const wrapperDisabled = shallow(<AccordionToggle {...disabledProps} />);
    wrapperDisabled.props().onClick();
    expect(mockToggleHandler).toHaveBeenCalledTimes(0);

    const wrapperEnabled = shallow(<AccordionToggle {...mockProps} />);
    wrapperEnabled.props().onClick();
    expect(mockToggleHandler).toHaveBeenCalled();
  });
});
