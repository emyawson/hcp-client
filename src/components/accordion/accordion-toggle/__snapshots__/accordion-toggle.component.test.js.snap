// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Toggle renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    disabled={false}
    id="abc-123"
    toggleHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "active": true,
      "children": Array [
        <styled.div
          id="AccordionTitle--abc-123"
        />,
        <styled.div
          active={true}
          disabled={false}
        >
          <Unknown
            active={false}
            direction="down"
            disabled={false}
            onClick={[Function]}
            size={24}
          />
        </styled.div>,
      ],
      "disabled": false,
      "isNested": false,
      "onClick": [Function],
      "onKeyPress": [Function],
      "role": "heading",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "id": "AccordionTitle--abc-123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": <Unknown
            active={false}
            direction="down"
            disabled={false}
            onClick={[Function]}
            size={24}
          />,
          "disabled": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "direction": "down",
            "disabled": false,
            "onClick": [Function],
            "size": 24,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": Array [
          <styled.div
            id="AccordionTitle--abc-123"
          />,
          <styled.div
            active={true}
            disabled={false}
          >
            <Unknown
              active={false}
              direction="down"
              disabled={false}
              onClick={[Function]}
              size={24}
            />
          </styled.div>,
        ],
        "disabled": false,
        "isNested": false,
        "onClick": [Function],
        "onKeyPress": [Function],
        "role": "heading",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "id": "AccordionTitle--abc-123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": <Unknown
              active={false}
              direction="down"
              disabled={false}
              onClick={[Function]}
              size={24}
            />,
            "disabled": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "direction": "down",
              "disabled": false,
              "onClick": [Function],
              "size": 24,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
